src/Microsoft.DotNet.TestFramework/TestAssetsManager.cs

```diff
@@ -107,12 +107,48 @@ public TestDirectory CreateTestDirectory([CallerMemberName] string callingMethod
 
         private string GetTestDestinationDirectoryPath(string testProjectName, string callingMethod, string identifier)
         {
-#if NET451
+// HACK: Make the TestAssetsManager create shorter paths for test assets.
+// This is a workaround for https://github.com/dotnet/coreclr/issues/5046
+// Remove this hack once the above issue is fixed.
+/*#if NET451
             string baseDirectory = AppDomain.CurrentDomain.BaseDirectory;
 #else
             string baseDirectory = AppContext.BaseDirectory;
+#endif*/
+
+            string path = Path.Combine(GetRepoRoot(), "bin", callingMethod + identifier, testProjectName);
```

Comment:
Also, shouldn't the testProjectName come before the callingMethod? So that all the tests for a project are co-located?